lib/connection.js

```diff
@@ -440,20 +446,15 @@ Connection.prototype._write = function (operation, streamId) {
   const self = this;
   this.writeQueue.push(operation, function writeCallback (err) {
     if (err) {
+      // There was a serialization error or the operation has already timed out or was cancelled
+      self._streamIds.push(streamId);
```

Comment:
In this case do we know if the write actually happened?  If not it makes sense to return the stream id, otherwise we shouldn't return it unless we get a response.

Reply:
I looked at the code and as far as I can tell the callback is invoked before we actually do the write, so if it were to fail it would be something like an encoding error so the write wouldn't have happened, so makes sense to return the stream id.